packages/common-all/src/utils/lookup.ts

```diff
@@ -65,6 +65,14 @@ export class NoteLookupUtils {
     return lastDotIndex < 0 ? "" : qs.slice(0, lastDotIndex + 1);
   };
 
+  static fetchRootResultsFromEngine = async (engine: DEngineClient) => {
```

Comment:
Can we use `ReducedDEngine` instead of `DEngineClient` here and whenever possible? This makes it such that the web engine can also work with the util.

Reply:
Good point. I'll keep that in mind moving forward